sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -442,17 +442,32 @@ class Analyzer(
         child: LogicalPlan): LogicalPlan = {
       val gid = AttributeReference(VirtualColumn.groupingIdName, IntegerType, false)()
 
+      val finalGroupByExpressions = if (groupByExprs == Nil) {
```

Comment:
Shouldn't we do this in the branch of `case x: GroupingSets if x.expressions.forall(_.resolved) =>`? I think this `constructAggregate` method is also used by other clauses like `Cube` and `Rollup`.

Reply:
Ok. Mind to add a comment on this like `SPARK-24424: this only happens for ANSI-SQL compliant syntax for GROUPING SET`?

Reply:
@viirya Sure will do.